src/basic/terminal-util.c

```diff
@@ -882,6 +893,52 @@ int terminal_set_size_fd(int fd, const char *ident, unsigned rows, unsigned cols
         return 0;
 }
 
+int proc_cmdline_tty_size(const char *tty, unsigned *ret_rows, unsigned *ret_cols) {
+        _cleanup_free_ char *rowskey = NULL, *rowsvalue = NULL, *colskey = NULL, *colsvalue = NULL;
+        unsigned rows = UINT_MAX, cols = UINT_MAX;
+        int r;
+
+        if (!ret_rows && !ret_cols)
+                return 0;
+
+        tty = skip_dev_prefix(tty);
```

Comment:
Maybe, `path_extract_filename()` is better?

Reply:
Hmm, I think `skip_dev_prefix()` better indicates the intent. We really don't expect ttys to be outside of /dev.

Reply:
That's not a strong opinion. Feel free to ignore it.